### ISSUE TITLE
Implement the regex check for roll numbers

### ISSUE THREAD

**Issue by icyflame** [icyflame/leprechaun#7]

**Comment by dibyadas** [comment]
Does this regex work?  
`[0-9]{2}[A-Z]{2}[0-9]{1,3}[A-Z]*[0-9]{2}`  And then post-process on it to check correct departments and micro courses. Ex - `16CH3PE34`

**Comment by icyflame** [comment]
No, we can't have any `*` characters, because we have an upper limit on the number of characters.

`13AB879THYFHYFHY09` matches your regex.

A liberal length limited regex can be: `[0-9]{2}[A-Z]{2}[0-9A-Z]{5}`

or `[0-9]{2}[A-Z]{2}[0-9]{1}[0-9A-Z]{4}`

or `[0-9]{2}[A-Z]{2}[0-9A-Z]{3}[0-9]{2}`

If you are interested in working on this, just open the PR with a good length limited PR. Any other discussion can continue there. :slightly_smiling_face: 

**Note:** In any case, even if someone adds a roll number that isn't real, they won't be able to verify unless they somehow get access to the leprechaun DB. But if they do manage to get fictional roll numbers verified, they can do some bad things with applications like voting systems. (Far-fetched scenario)

**Comment by dibyadas** [comment]
Oh. Yeah, Missed that. Cool. I'll open a PR. Thanks!
